### PR TITLE
[API][Order] Fulfilling orders

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

Part of https://github.com/Sylius/Sylius/issues/11250

### CODE REVIEW

File: src/Sylius/Behat/Context/Api/Admin/ManagingOrdersContext.php [hunk 1]
```diff
@@ -76,6 +77,30 @@ public function iCancelThisOrder(OrderInterface $order): void
...
+        );
+    }
+
+    /**
+     * @When /^I ship (this order)$/
+     */
+    public function iShipThisOrder(OrderInterface $order): void
+    {
+        $this->client->customEndPoint(
+            sprintf('/shipments/%s/ship', (string) $order->getShipments()->first()->getId()),
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/Admin/ManagingOrdersContext.php]: That's why we need some possibility to gather links and available actions :(

Comment (lchrusciel) [reply src/Sylius/Behat/Context/Api/Admin/ManagingOrdersContext.php]: Another question is, do we want to expose these actions also as sub resources of orders (what we are doing in admin panel). Perhaps not?

**Comment by lchrusciel** [comment]
Part of #11250

File: src/Sylius/Behat/Context/Api/Admin/ManagingOrdersContext.php (hunk 1 above)
Comment (Zales0123) [reply src/Sylius/Behat/Context/Api/Admin/ManagingOrdersContext.php]: Perhaps... I don't know 🤷‍♂. On the one hand, It seems reasonable to access shipments via orders. You check out the orders list (`orders/`), then the specific order (`orders/000111`), you want to see its shipments (`orders/000111/shipments`), especially one of them (`orders/000111/shipments/12`), and to ship it (`orders/000111/shipments/12/dancer`) 💃 But it seems like a long way and probably an improvement to the base as it is right now 🖖

**Comment by lchrusciel** [comment]
part of #11250

File: src/Sylius/Behat/Client/Request.php [hunk 2]
```diff
@@ -100,7 +100,11 @@ public static function transition(string $resource, string $id, string $transiti
 
     public static function custom(string $url, string $method, string $token): RequestInterface
     {
-        return new self($url, $method, ['HTTP_Authorization' => 'Bearer ' . $token]);
+        return new self(
```
Comment (lchrusciel) [inline src/Sylius/Behat/Client/Request.php]: Probably not relevant to this PR

File: src/Sylius/Behat/Context/Api/Admin/ManagingOrdersContext.php [hunk 3]
```diff
@@ -76,6 +89,28 @@ public function iCancelThisOrder(OrderInterface $order): void
         );
     }
 
+    /**
+     * @When /^I mark (this order) as paid$/
+     */
+    public function iMarkThisOrderAsAPaid(OrderInterface $order): void
+    {
+        $this->paymentsClient->applyTransition(
+            (string) $order->getLastPayment()->getId(),
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/Admin/ManagingOrdersContext.php]: Something that we should consider is to test these actions when we will access them from the order show response

**Comment by lchrusciel** [comment]
Thanks, Adam! :1st_place_medal:

**Pull request merged**